src/o365/graph/commands/reports/reports-teams-device-usage-user-detail.ts

```diff
@@ -0,0 +1,153 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import request from '../../../../request';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+const maxDays = 28;
+const maxDate: Date = new Date();
+maxDate.setDate(maxDate.getDate() - maxDays);
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  period?: string;
+  date?: string;
+}
+
+class GraphReportsTeamsDeviceUsageUserDetailCommand extends GraphItemsListCommand<any> {
+  
+  public get name(): string {
+    return `${commands.REPORTS_TEAMS_DEVICE_USAGE_USER_DETAIL}`;
+  }
+
+  public get description(): string {
+    return 'Gets detail about Microsoft Teams device usage by user';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.period = typeof args.options.period !== 'undefined';
+    telemetryProps.date = typeof args.options.date !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const periodParameter: string = args.options.period ? `getTeamsDeviceUsageUserDetail(period='${encodeURIComponent(args.options.period)}')` : '';
+    const dateParameter: string = args.options.date ? `getTeamsDeviceUsageUserDetail(date=${encodeURIComponent(args.options.date)})` : '';
+
+    let endpoint: string = args.options.period ? `${auth.service.resource}/v1.0/reports/${periodParameter}` : `${auth.service.resource}/v1.0/reports/${dateParameter}`;
```

Comment:
Since the first portion of the URL is the same, let's not repeat it and change it to something like:

```ts
const endpoint: string = ``${auth.service.resource}/v1.0/reports/${(args.options.period ? periodParameter : dateParameter )}`;
```

Reply:
Fixed and ready for review.